### DISCUSSION TITLE
Help to install dsniff and/or arpspoof on Pineapple cloner

### DISCUSSION THREAD

**Discussion by blackrosevn** [xchwarze/wifi-pineapple-cloner#7] [Q&A]
Hello there,

 I would like to ask if someone can help and guide me how to install the arpspoof and/or dsniff on the Pineapple cloner.

I did try and find on the internet but could not find a way.

Here is the error message:

**Unknown package 'arpspoof'.
Collected errors:
 * opkg_install_cmd: Cannot install package arpspoof**

Thanks

**Comment by xchwarze** [comment]
If you run opkg update before and it doesn't install them, then they are not in any of the download repos.
If so you have to find the apk and install it manually.

**Reply by blackrosevn** [reply]
I have 2 routers: one is with mips_24kc, and they are in default repos to install. But the other one with mipsel_24kc (Xiaomi R3G), they do not have in the repos, even I have been trying to find a lot of time on Google, but it seems nothing for this mipsel_24k.

Can you help to include an ipk in the repos or show me a link to download that ipk?

Thanks

**Reply by xchwarze** [reply]
```
Package: dsniff
Version: 2.4b1-2
Depends: libc, libpcap1, libnids, libnet0, libopenssl1.1, libgdbm, libtirpc
Source: extra_feeds/community/network/dsniff
SourceName: dsniff
Section: net
Architecture: mips_24kc
Installed-Size: 57039
Description:  A collection of tools for network auditing and penetration testing.
```

```
Package: arpspoof
Version: 2.4b1-2
Depends: libc, libpcap1, libnids, libnet0, libopenssl1.1, libgdbm, libtirpc
Source: extra_feeds/community/network/dsniff
SourceName: arpspoof
Section: net
Architecture: mips_24kc
Installed-Size: 2758
Description:  Arpspoof package.
```

None of them mention who maintains them, so I guess they were compiled by hak5.
At worst you would have to find the sources and compile it for openwrt 19.

**Comment by blackrosevn** [comment]
After searching I realised that the Devices I have are mipsel_24kc without repos support for dsniff and arpspoof (what you shared is for mips_24kc).

So I bought a Mikrotik RB951G-2HND - mips_24kc, then I am going to build the cloner firmware for this device. But it seems my device is not yet added to your support code.

Could you please guide me how build with the sample command "tools/builder.sh mips universal imagebuilder/19.07.7-ar71xx-generic gl-ar750s" => Which code for my device instead of gl-ar750s?

**Reply by blackrosevn** [reply]
Can anyone help me to find a cloner pineapple firmware for Mikrotik RB951G-2HND? Or how to build it from the cloner git? I could not find a way neither.

I did search in the target list but there is nothing relating to the Mikrotik RB951G-2HND.

And I am already done with Openwrt running on the device. Just one more step to make it to be pineapple.

So please check and help.

Thanks!

**Reply by xchwarze** [reply]
I checked and it's not in the "generic" variant because apparently it's a separate variant.
https://mirror-01.infra.openwrt.org/releases/19.07.7/targets/ar71xx/

You should add that new variant in the supported variant array, its name would be "ar71xx-mikrotik"
https://github.com/xchwarze/wifi-pineapple-cloner/blob/master/tools/dependencies-install.sh#L7

And the profile name should be vmlinux (I haven't downloaded that imagebuilder, if you want to attach what `make info` returns so I can see what values are there).
Remember to follow the steps in the OpenWrt manual and convert the router to the OpenWrt platform first: https://openwrt.org/toh/mikrotik/rb951g_2hnd

**Reply by blackrosevn** [reply]
Thank you @xchwarze !

I finally can generate the firmware for my device following your comments.

**Reply by xchwarze** [reply]
@blackrosevn whats is the profile name? vmlinux?

**Reply by blackrosevn** [reply]
I chose nand-large as my device has 128MB flash.

**Reply by th3cha1rman** [reply]
@blackrosevn did it work?

**Comment by bad-antics** [comment]
Installing dsniff/arpspoof on a Pineapple cloner can be tricky due to OpenWrt package availability. Here is how:

## Method 1: opkg (if available)
```bash
opkg update
opkg install dsniff
```
If `dsniff` is not in the default feeds, you may need to add the full OpenWrt package repository.

## Method 2: Cross-compile from source
If opkg does not have dsniff:
```bash
# On your build machine (not the router)
git clone https://github.com/openwrt/packages
# Find dsniff in net/dsniff
# Build with the OpenWrt SDK matching your router architecture
```

## Method 3: Use ettercap instead
Ettercap is often easier to get running on OpenWrt and does everything arpspoof does plus more:
```bash
opkg update
opkg install ettercap
# ARP poisoning
ettercap -T -M arp:remote /target_ip// /gateway_ip//
```

## Method 4: Manual arpspoof with arping
If you just need basic ARP spoofing:
```bash
# Install arping
opkg install iputils-arping

# Manual ARP poison (basic script)
while true; do
  arping -U -I br-lan -s <gateway_ip> <target_ip>
  sleep 1
done
```

## Important
- Enable IP forwarding: `echo 1 > /proc/sys/net/ipv4/ip_forward`
- These tools are for **authorized security testing only**
- I maintain [ARP spoofing payloads](https://github.com/bad-antics/hak5-pineapple/tree/main/nullsec-suite/ARPSpoof) for the Pineapple if you want a ready-made solution
